meta/core.json

```
@@ -43,9 +44,11 @@
             "type": "string",
             "format": "iri"
         },
-        "iriReferenceString": {
+        "iriReferenceToSchemaString": {
             "type": "string",
-            "format": "iri-reference"
+            "format": "iri-reference",
+            "$comment": "any fragment must be empty, or match anchor or json-pointer syntax",
+            "pattern": "^[^#]*(#([A-Za-z_][-A-Za-z0-9.:_]*|/([^~]|~[01])*))?$"
```

```suggestion
            "pattern": "^[^#]*(#([A-Za-z_][-A-Za-z0-9.:_]*|(/([^~/]|~[01])*)*))?$"
```

The second half of the fragment match, for JSON Pointer, won't match the empty pointer (`$ref: "#"`, `$ref: "foo#"`). The suggested change follows the [JSON Pointer ABNF](https://www.rfc-editor.org/rfc/rfc6901.html#section-3).

This is still incomplete though. Fragments may contain %xx encoded characters. In a pointer their use may be necessary, but in either a pointer or anchor they are expected to be treated like the characters they represent, even when those characters don't actually need to be %-encoded.

```suggestion
            "pattern": "^[^#]*(#(([A-Za-z_]|%[A-Fa-f0-9]{2})([-A-Za-z0-9.:_]|%[A-Fa-f0-9]{2})*|((/|%2[fF])([^~/%]|~[01]|%[A-Fa-f0-9]{2})*)*))?$"
```

Not too pretty, but I think it allows anything that's valid. It won't reject some invalid anchors or pointers if they're %-encoded - it would be possible to express the same character ranges as allowed combinations of %-encoded digits, but I'm not masochistic enough to do it.

I'm a little uncertain describing a uri fragment with a regex isn't more trouble than it's worth.